src/pages/index.js

```diff
@@ -1,155 +1,319 @@
 /*                                Card Section                                */
 /* -------------------------------------------------------------------------- */
-function renderCard(data) {
+function handleConfirmationOpen(cardId, cardElement) {
+  deletePopup.open(cardId, cardElement);
+}
+
+const existingCardIds = new Set();
+
+function createCard(data) {
+  if (existingCardIds.has(data._id)) {
+    return null; // Card already exists, do not create a duplicate
+  }
+
   const card = new Card(
     data,
-    (imageData) => {
-      imagePopup.open(imageData);
-    },
-    "#card-template"
+    handleImageClick,
+    "#card-template",
+    api,
+    handleDelete
   );
-  return card.getView();
+  const cardElement = card.getView();
+  cardElement.dataset.id = data._id;
+  existingCardIds.add(data._id);
+  return cardElement;
+}
+
+function handleDelete(cardId, cardElement) {
+  handleConfirmationOpen(cardId, cardElement);
 }
 
-// Initialize card section
+const deletePopup = new PopupWithConfirm(
+  "#verify__card__delete",
+  (cardId, cardElement) => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardElement.remove();
+        deletePopup.close();
+        existingCardIds.delete(cardId); // Remove from Set when card is deleted
+      })
+      .catch((error) => {
+        console.error("Error deleting card:", error);
+      });
+  }
+);
+
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (data) => {
-      // Create a new card
-      const cardElement = renderCard(data);
-
-      // Display each card
-      cardSection.addItem(cardElement);
+    items: [],
+    renderer: (item) => {
+      const cardElement = createCard(item);
+      if (cardElement) {
+        cardSection.addItem(cardElement);
+      }
     },
   },
-  formSettings.cardList
+  ".cards__list"
 );
 
-// Enable validation for all forms
+const addCardPopup = new PopupWithForm("#modalAddCard", (data) => {
+  addCardPopup.showLoading(); // Show loading before API call
+  api
+    .renderCard(data.name, data.link)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      if (cardElement) {
+        cardSection.addItem(cardElement);
 
-/* -------------------------------------------------------------------------- */
-/*                                  Add Crad  Form                            */
-/* -------------------------------------------------------------------------- */
-
-// Create the add form instance
-const addCardPopup = new PopupWithForm(formSettings.modalAddCard, (data) => {
-  // Create a new card
-  const newCard = renderCard(data);
-  // Close the add form
-  addCardPopup.close();
-  // Add the new card to the section
-  cardSection.addItem(newCard);
+        const formName = document
+          .querySelector("#modalAddCard .modal__form")
+          .getAttribute("name");
+        formValidator[formName].resetValidation();
+        addCardPopup.close();
```

Comment:
* You need to **`disable`** the submit button in the `card popup` if the inputs are **empty** when you open it. It’s needed because I can click the button and add an empty card but it should **not** happen. After the first submitting the button gets activated and you need to **deactivate** it again. Even if I can't add an empty card the button should look **disabled** (gray) anyway and with attribute `disabled`.


You need to close the modal **before** you call `resetValidation` because you need to reset the form first and them toggle the button


```js
        addCardPopup.close();  // first close
        const formName = addCardForm.getAttribute("name");
        formValidator[formName].resetValidation();

```